apis/v1beta1/collector_webhook_test.go

```diff
@@ -1244,20 +1304,58 @@ func TestOTELColValidatingWebhook(t *testing.T) {
 			},
 			expectedErr: "the OpenTelemetry Collector mode is set to statefulset, which does not support the attribute 'deploymentUpdateStrategy'",
 		},
+		{
+			name: "missing port for ingress type",
```

Comment:
The changes in this file are mostly package fixes. This is an added test to check that webhook reconciliation errors are working properly.